Source/UI/Message/FeedStrategyMessageList.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  FeedStrategyMessageList.swift
+//  Planetary
+//
+//  Created by Martin Dutra on 19/12/22.
+//  Copyright © 2022 Verse Communications Inc. All rights reserved.
+//
+
+import CrashReporting
+import Foundation
+import Logger
+
+class FeedStrategyMessageList: MessageList {
+    let strategy: FeedStrategy
+    let bot: Bot
+
+    init(strategy: FeedStrategy, bot: Bot) {
+        self.strategy = strategy
+        self.bot = bot
+    }
+
+    @MainActor
+    @Published
+    var cache: [Message]?
+
+    @Published
+    var isLoadingMore = false
+
+    @Published
+    var isLoadingFromScratch = false
+
+    @Published
+    var errorMessage: String?
+
+    /// The number of loaded pages
+    @Published
+    var pages = 0
+
+    @MainActor
+    var isEmpty: Bool {
+        if let cache = cache {
+            return cache.count == 0
+        }
+        return false
+    }
+
+    /// The size of each page
+    private let pageSize = 50
+
+    /// The number of messages loaded
+    private var offset = 0
+
+    /// If true, we know there are no more pages to be loaded
+    private var noMoreMessages = false
+
+    @MainActor
+    func loadFromScratch() async {
```

Comment:
It looks to me like `loadFromScratch()` could just call `loadMore()` and it would do pretty much the same thing. The only difference I see is that `loadFromScratch()` shows and error message to the user on failure and `loadMore()` doesn't. Maybe we could upgrade `loadMore()` to show an error message and then get rid of `loadFromScratch()`. Or am I missing something?

If we do need to keep both functions around, let's write some comments (probably on InfiniteList) about what the different expectations are for someone implementing them in the future.

Reply:
I added comments in the protocol that describe the minimum requirements. Implementations could use a single function if they find that both operations are the same. However, the FeedMessage implementation I did here has a lot of differences between loadFromScratch and loadMore.